scripts/httpd/upload.cgi

```diff
@@ -6,56 +6,59 @@
 #
 # License: GPL
 
+import cgi
+import errno
+import grp
+import hashlib
 import os
 import sys
-import cgi
 import tempfile
-import grp
-import pwd
-import syslog
-import smtplib
-import errno
 
 import fedmsg
 import fedmsg.config
 
-from ConfigParser import ConfigParser
-
-from email import Header, Utils
-try:
-    from email.mime.text import MIMEText
-except ImportError:
-    from email.MIMEText import MIMEText
-
-import hashlib
 
 # Reading buffer size
 BUFFER_SIZE = 4096
 
 # We check modules exist from this dircetory
-GITREPO = '/var/lib/dist-git/git/rpms'
+GITREPO = '/srv/git/repositories'
 
 # Lookaside cache directory
-CACHE_DIR = '/var/lib/dist-git/cache/lookaside/pkgs'
+CACHE_DIR = '/srv/cache/lookaside/pkgs'
 
 # Fedora Packager Group
 PACKAGER_GROUP = 'packager'
 
-# dist git configuration file
-CONFIG_FILE = "/etc/dist-git/dist-git.conf"
 
-def send_error(text):
+def send_error(text, status='500 Internal Server Error'):
+    """Send an error back to the client
+
+    This ensures that the client will get a proper error, including the HTTP
+    status code, so that it can handle problems appropriately.
+
+    Args:
+        text (str): The error message to send the client
+        status (str, optional): The HTTP status code to return to the client.
+    """
+    print 'Status: %s' % status
+    print 'Content-type: text/plain'
+    print
     print text
-    sys.exit(1)
+    sys.exit(0)
```

Comment:
ACK, but it deserves a comment that cgi exit status is ignored by apache (i think).